### PR TITLE
Fixing logs

### PR DESCRIPTION
**Pull request by AnomalRoil**
Backport from master, see #920

### CODE REVIEW

**Review by willscott — APPROVED** [review]

File: chain/beacon/node.go [hunk 1]
```diff
@@ -105,31 +105,35 @@ func (h *Handler) ProcessPartialBeacon(c context.Context, p *proto.PartialBeacon
...
 
 	msg := chain.Message(p.GetRound(), p.GetPreviousSig())
-	// XXX Remove that evaluation - find another way to show the current dist.
-	// key being used
-	shortPub := h.crypto.GetPub().Eval(1).V.String()[14:19]
+	idx, _ := key.Scheme.IndexOf(p.GetPartialSig())
+	if idx < 0 {
+		return nil, fmt.Errorf("invalid index %d in partial with msg %v", idx, msg)
+	}
+	nodeName := h.crypto.GetGroup().Node(uint32(idx)).Identity.Address()
```
Comment (nikkolasg) [inline chain/beacon/node.go:112]: Node can return nil

File: chain/beacon/node.go [hunk 2]
```diff
@@ -105,31 +105,35 @@ func (h *Handler) ProcessPartialBeacon(c context.Context, p *proto.PartialBeacon
 	}
 
 	msg := chain.Message(p.GetRound(), p.GetPreviousSig())
-	// XXX Remove that evaluation - find another way to show the current dist.
-	// key being used
-	shortPub := h.crypto.GetPub().Eval(1).V.String()[14:19]
+	idx, _ := key.Scheme.IndexOf(p.GetPartialSig())
```
Comment (nikkolasg) [inline chain/beacon/node.go:108]: IndexOf returns an error if index is not parsable as integer

**Comment by AnomalRoil** [comment]
@willscott @nikkolasg The release 1.3 branch seems to have some issues with its tests, https://github.com/drand/drand/commit/6405db0259684f8b1d882d9941f34b5802ef084c was already not passing. 
Should we go ahead and merge this or should I spend time trying to patch the tests on this old release?

**Comment by willscott** [comment]
* the 1.3 branch is where current logs are coming from.
* it uses a different logger from what's in master. worth testing manually the change and making sure you get output as expected from the changed lines
* assuming it's just a flaky test, i think it's fine not to spend the effort back-porting the test fix-up to the branch

**Comment by nikkolasg** [comment]
I am merging it because this is only changing logs and we need to advance on this - the flaky tests were already present before this change anyway.

**Pull request merged**